Sprint 07

Kolesnikov-Eugene opened the pull request.

Kolesnikov-Eugene committed:
Added Unit-test of moviesLoader function.
Added UITest  of Yes, No buttons, and alert presentation and dismissal

Kolesnikov-Eugene committed:
added MovieQuizPresenter class. func convert moved to MovieQuizPresenter

Kolesnikov-Eugene committed:
added functions yesButtonPressed and noButtonPressed to MovieQuizPresenter

Kolesnikov-Eugene committed:
Бизнес-логика отделена от представления и полностью вынесена в класс MovieQuizPresenter

Kolesnikov-Eugene committed:
Redesigned logic of proceedToNextQuestionOrResult method, added createResult method.

Kolesnikov-Eugene committed:
final version

Kolesnikov-Eugene committed:
The first test removed from project (addition, subtraction, etc).

protos17 commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,139 @@
+import UIKit
+
+final class MovieQuizPresenter {
+    private weak var vc: MovieQuizViewControllerProtocol?
```
Рекомендую переименовать во viewController, потому что принято называть переменные больше, чем двумя буквами, на что линтер может ругаться, который подключен в большинстве продакшен проектов

protos17 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -61,101 +40,51 @@ final class MovieQuizViewController: UIViewController {
-    func didLoadDataFromServer() {
+extension MovieQuizViewController: MovieQuizViewControllerProtocol {
+    func hideLoadingIndicator() {
         activityIndicator.isHidden = true
```
Рекомендую использовать свойство hidesWhenStopped вместо ручного управления свойством isHidden

protos17 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -61,101 +40,51 @@ final class MovieQuizViewController: UIViewController {
-        }
-        alertPresenter?.show(alert: model)
+    func showLoadingIndicator() {
+        activityIndicator.isHidden = false
```
И здесь тогда не понадобится самостоятельно его отображать

protos17 commented on MovieQuizTests/ArrayTests.swift:
```diff
@@ -0,0 +1,31 @@
+import XCTest
+@testable import MovieQuiz
+
+class ArrayTests: XCTestCase {
```
Для тестовых классов также рекомендую указывать как final

protos17 commented on MovieQuizTests/ArrayTests.swift:
```diff
@@ -0,0 +1,31 @@
+class ArrayTests: XCTestCase {
+    func testGetValueInRange() throws {
+        // Given
+        let arr = [1, 1, 2, 3, 5]
```
Для тестов рекомендую использовать уникальные значения, потому что arr[safe: 0] и arr[safe: 1] дадут один результат

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+    }
+    
+    func testAlertPresentation() {
+        sleep(5)
```
Слишком большое время ожидания, рекомендую сократить

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+        var taps = 0
+        while taps < 10 {
+            app.buttons["Yes"].tap()
+            sleep(3)
```
У тебя через секунду показывается следующий вопрос, можно сделать, чтобы через секунду и нажимался

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+        var taps = 0
+        while taps < 10 {
+            app.buttons["Yes"].tap()
+            sleep(3)
```
Здесь тоже рекомендую сократить до секунды

Kolesnikov-Eugene committed:
changed activityIndicator presentation (using hidesWhenStopped property instead .isHidden property).
vc property renamed to viewController in MovieQuizPresenter and AlertPresenter classes.
